lib/snapshot.go

```diff
@@ -443,18 +444,29 @@ func NewSnapshot(
 	//   Either way, it means our snapshot was compromised and we need to recompute it as described
 	//   in the previous bullet.
 	shouldRestart := false
+	isChecksumIssue := false
+	// @diamondhands - I think the only operations tracked by the state semaphore are (1) checksum add, (2) checksum
+	// remove, (3) ProcessChunk, (4) ChecksumPrint, (5) Exit. So the only one operation that isn't checksum related
+	// is the ProcessChunk operation. However, I think we can get away with this being considered a "checksum" issue
+	// and not restarting the node (which causes a panic). If you have a node that didn't finish processing chunks,
+	// just start over again.
```

Comment:
I would add just for extra clarity:

However, I think we can get away with this being considered a "checksum" issue **because ProcessChunk only happens during initial sync**